### PR TITLE
documentation related update

### PR DESCRIPTION
**Pull request by JaydenLiang**

### CODE REVIEW

**Comment by JaydenLiang** [comment]
Hi @ilee-fortinet , I invited you to review my PR here.

File: templates/autoscale-new-vpc.template.yaml [hunk 1]
```diff
@@ -6,510 +6,337 @@ Description: >-
...
+        Type: Number
+    FgtAsgMaxSizePayg:
         ConstraintDescription: must be a valid number not less than 0.
-        Description: >-
-            Minimum number of FortiGate instances in the BYOL Auto Scaling
-            group. For specific use cases, set to 0 for On-Demand-only, and >= 2
-            for BYOL-only or hybrid licensing.
-    FgtAsgMaxSizeByol:
+        Default: 6
+        Description: Maximum number of FortiGate instances in the On-Demand Auto Scaling group. For specific use cases, set to 0 for BYOL-only, >= 2 for On-Demand-only, and >= 0 for hybrid licensing. This number must be greater than or equal to the Minimum group size (On-Demand).
```
Comment (ilee-fortinet) [inline templates/autoscale-new-vpc.template.yaml:68]:
```suggestion
        Description: Maximum number of FortiGate instances in the On-Demand Auto Scaling group. For specific use cases, set to 0 for BYOL-only, >= 2 for On-Demand-only, and >= 0 for hybrid licensing. This number must be greater than or equal to the minimum group size (On-Demand).
```

File: templates/autoscale-new-vpc.template.yaml [hunk 2]
```diff
@@ -6,510 +6,337 @@ Description: >-
...
-        Description: >-
-            The threshold (in percentage) for the FortiGate Auto Scaling group
-            to scale-out (add) 1 instance. Minimum is 1. Maximum is 100.
-    FortiGatePskSecret:
-        Type: String
+        MinValue: 1
+        Type: Number
+    FortiAnalyzerAutoscaleAdminPassword:
+        Default: ''
+        Description: The password for the 'Autoscale admin username'. The password must conform to the FortiAnalyzer password policy and have a min length of 8 and a max length 128. If you need to enable KMS encryption, refer to the documentation.
```
Comment (ilee-fortinet) [inline templates/autoscale-new-vpc.template.yaml:99]:
```suggestion
        Description: The password for the 'Autoscale admin username'. The password must conform to the FortiAnalyzer password policy and have a minimum length of 8 and a maximum length of 128. If you need to enable KMS encryption, refer to the documentation.
```

File: templates/autoscale-new-vpc.template.yaml [hunk 3]
```diff
@@ -6,510 +6,337 @@ Description: >-
...
-            between sending heartbeat requests to the Autoscale handler. Minimum
-            is 30. Maximum is 90.
-    HeartBeatLossCount:
+        MinValue: 30
         Type: Number
+    HeartBeatLossCount:
+        ConstraintDescription: must be a valid number between 1 and 65535.
         Default: 10
-        MinValue: 1
+        Description: Number of consecutively lost heartbeats. When the Heartbeat Loss Count has been reached, the VM is deemed unhealthy and fail-over activities will commence.
```
Comment (ilee-fortinet) [inline templates/autoscale-new-vpc.template.yaml:207]:
```suggestion
        Description: Maximum number of consecutively lost heartbeats. When the Heartbeat loss count is reached, the VM is deemed unhealthy and failover activities commence.
```

File: templates/autoscale-new-vpc.template.yaml [hunk 4]
```diff
@@ -6,510 +6,337 @@ Description: >-
...
-        Default: 2
-        MinValue: 0
-        Description: >-
-            The maximum amount of time (in seconds) allowed for network latency
-            of the FortiGate heartbeat arriving at the Autoscale handler.
-            Minimum is 0.
-    GetLicenseGracePeriod:
+    InternalLoadBalancerDnsName:
+        Default: ''
+        Description: Optional. Specify the DNS Name of an existing internal load balancer used to route traffic from a FortiGate to targets in a specified target group. Leave it blank if you don't use an existing load balancer.
```
Comment (ilee-fortinet) [inline templates/autoscale-new-vpc.template.yaml:213]:
```suggestion
        Description: Optional. Specify the DNS name of an existing internal load balancer used to route traffic from a FortiGate to targets in a specified target group. Leave it blank if you do not use an existing load balancer.
```

File: templates/autoscale-new-vpc.template.yaml [hunk 5]
```diff
@@ -6,510 +6,337 @@ Description: >-
...
+        Type: String
+    InternalLoadBalancingOptions:
+        AllowedValues:
+            - add a new internal load balancer
+            - use a load balancer specified below
+            - do not need one
+        ConstraintDescription: must choose from the provided options.
+        Default: add a new internal load balancer
+        Description: (Optional) Add a predefined Elastic Load Balancer (ELB) to route traffic to web service in the private subnets. You can optionally use your own one or decide to not need one.
```
Comment (ilee-fortinet) [inline templates/autoscale-new-vpc.template.yaml:222]:
```suggestion
        Description: (Optional) Add a predefined Elastic Load Balancer (ELB) to route traffic to web service in the private subnets. You can optionally use your own or not use one.
```

File: templates/autoscale-new-vpc.template.yaml [hunk 6]
```diff
@@ -6,510 +6,337 @@ Description: >-
...
+    InternalTargetGroupHealthCheckPath:
+        AllowedPattern: ^/[0-9a-zA-Z-/]*$
+        ConstraintDescription: This path must begin with a '/' character, and can be at most 1024 characters in length.
+        Default: /
+        Description: Optional. The destination path for health checks. This path must begin with a '/' character, and can be at most 1024 characters in length.
+        MaxLength: 1024
+        Type: String
+    KeyPairName:
+        ConstraintDescription: must specify an admin access key pair for FortiGate instances.
+        Description: Amazon EC2 Key Pair for admin access.
```
Comment (ilee-fortinet) [inline templates/autoscale-new-vpc.template.yaml:233]:
```suggestion
        Description: Amazon EC2 key pair for admin access.
```

File: templates/autoscale-new-vpc.template.yaml [hunk 7]
```diff
@@ -6,510 +6,337 @@ Description: >-
...
-        Default: 'no'
-        AllowedValues:
-            - 'yes'
-            - 'no'
-        ConstraintDescription: must choose from the provided options.
-        Description: Will use a custom S3 location for custom assets.
-    CustomAssetContainer:
+    ResourceTagPrefix:
+        AllowedPattern: '[0-9a-zA-Z@.#-]+'
+        ConstraintDescription: Resource tag prefix can include numbers, lowercase letters, uppercase letters, ampersat(@) , hyphens (-), period (.), and hash (#). Max length is 50.
```
Comment (ilee-fortinet) [inline templates/autoscale-new-vpc.template.yaml:297]:
```suggestion
        ConstraintDescription: Resource tag prefix can include numbers, lowercase letters, uppercase letters, ampersats (@) , hyphens (-), periods (.), and hashes (#). Max length is 50.
```

File: templates/autoscale-new-vpc.template.yaml [hunk 8]
```diff
@@ -6,510 +6,337 @@ Description: >-
...
+        Description: The ResourceGroup Tag Key used on all resources and as the name prefix of all applicable resources. Can only contain numbers, lowercase letters, uppercase letters, ampersat(@), hyphens (-), period (.), and hash (#). Max length is 50.
```
Comment (ilee-fortinet) [inline templates/autoscale-new-vpc.template.yaml:298]:
```suggestion
        Description: The ResourceGroup Tag Key used on all resources and as the name prefix of all applicable resources. Can only contain numbers, lowercase letters, uppercase letters, ampersats (@), hyphens (-), periods (.), and hashes (#). Max length is 50.
```

File: templates/autoscale-new-vpc.template.yaml [hunk 9]
```diff
@@ -6,510 +6,337 @@ Description: >-
...
+        MaxLength: 50
         Type: String
-        Default: ''
-        Description: >-
-            The name of the S3 bucket which contains your custom assets.
-            Required if 'use custom asset location' is set to 'yes'.
-    CustomAssetDirectory:
+    S3BucketName:
+        AllowedPattern: ^[0-9a-zA-Z]+([0-9a-zA-Z-.]*[0-9a-zA-Z])*$
+        ConstraintDescription: Deployment resource S3 bucket name can include numbers, lowercase letters, uppercase letters, period (.), and hyphens (-). It cannot start or end with a hyphen (-).
```
Comment (ilee-fortinet) [inline templates/autoscale-new-vpc.template.yaml:303]:
```suggestion
        ConstraintDescription: Deployment resource S3 bucket name can include numbers, lowercase letters, uppercase letters, periods (.), and hyphens (-). It cannot start or end with a hyphen (-).
```

File: templates/autoscale-new-vpc.template.yaml [hunk 10]
```diff
@@ -6,510 +6,337 @@ Description: >-
...
+        Description: Name of the S3 bucket that contains the FortiGate Autoscale deployment package. Can only contain numbers, lowercase letters, uppercase letters, period (.), and hyphens (-). It cannot start or end with a hyphen (-).
```
Comment (ilee-fortinet) [inline templates/autoscale-new-vpc.template.yaml:304]:
```suggestion
        Description: Name of the S3 bucket that contains the FortiGate Autoscale deployment package. Can only contain numbers, lowercase letters, uppercase letters, periods (.), and hyphens (-). It cannot start or end with a hyphen (-).
```

File: templates/autoscale-new-vpc.template.yaml [hunk 11]
```diff
@@ -6,510 +6,337 @@ Description: >-
...
-        Description: >-
-            The sub path within the 'custom asset container' which serves as the
-            top level directory of all your custom assets. If 'use custom asset
-            location' is set to 'yes', and this value is left empty, the 'custom
-            asset container' will serve as the top level directory.
-    FortiAnalyzerIntegrationOptions:
+    S3KeyPrefix:
+        AllowedPattern: ^$|^[0-9a-zA-Z-./]*/$
+        ConstraintDescription: Deployment package S3 key prefix can include numbers, lowercase letters, uppercase letters, hyphens (-), period (.), and forward slash (/). If provided, it must end with a forward slash (/).
+        Description: Name of the S3 folder that stores the FortiGate Autoscale deployment resources. Can only contain numbers, lowercase letters, uppercase letters, hyphens (-), period (.), and forward slashes (/). If provided, it must end with a forward slash (/).
```
Comment (ilee-fortinet) [inline templates/autoscale-new-vpc.template.yaml:311]:
```suggestion
        Description: Name of the S3 folder that stores the FortiGate Autoscale deployment resources. Can only contain numbers, lowercase letters, uppercase letters, hyphens (-), periods (.), and forward slashes (/). If provided, it must end with a forward slash (/).
```

File: templates/autoscale-tgw-new-vpc.template.yaml [hunk 12]
```diff
@@ -7,362 +7,119 @@ Description: >-
...
-        Description: >-
-            Name of the S3 bucket that contains the FortiGate Autoscale
-            deployment package. Can only contain numbers, lowercase letters,
-            uppercase letters, period (.), and hyphens (-). It cannot start or
-            end with a hyphen (-).
-    S3KeyPrefix:
+    AutoscaleNotificationSubscriberEmail:
+        AllowedPattern: ^$|^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$
+        ConstraintDescription: must be a valid email address.
+        Description: The email address (AWS SNS Topic subscriber) to receive Autoscale notifications. If provided, the template can only accept one email address. An email will be sent to the address to confirm the subscription.
```
Comment (ilee-fortinet) [inline templates/autoscale-tgw-new-vpc.template.yaml:13]:
```suggestion
        Description: The email address (AWS SNS topic subscriber) to receive Autoscale notifications. If provided, the template can only accept one email address. An email is sent to the address to confirm the subscription.
```

File: templates/autoscale-tgw-new-vpc.template.yaml [hunk 13]
```diff
@@ -7,362 +7,119 @@ Description: >-
...
-        ConstraintDescription: must be a valid CIDR block format.
-        Description: >-
-            The CIDR block for the subnet located in Availability Zone 1 where
-            the FortiGate Autoscale instances will be deployed to.
-    PublicSubnet2Cidr:
+        Description: The list of Availability Zones to use for the subnets in the VPC. The FortiGate Autoscale solution uses two Availability Zones from your list and preserves the logical order you specify.
+        Type: List<AWS::EC2::AvailabilityZone::Name>
+    BgpAsn:
+        Default: 65000
+        Description: The Border Gateway Protocol (BGP) Autonomous System Number (ASN) of the Customer Gateway of each FortiGate-VM instance in the Auto Scaling group. This value ranges from 64512 to 65534.
```
Comment (ilee-fortinet) [inline templates/autoscale-tgw-new-vpc.template.yaml:20]:
```suggestion
        Description: The Border Gateway Protocol (BGP) Autonomous System Number (ASN) of the customer gateway of each FortiGate-VM instance in the Auto Scaling group. This value ranges from 64512 to 65534.
```

File: templates/autoscale-tgw-new-vpc.template.yaml [hunk 14]
```diff
@@ -7,362 +7,119 @@ Description: >-
...
-        AllowedPattern: >-
-            ^(([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5]){1}(\/([0-9]|[1-2][0-9]|3[0-2]))?$
-        ConstraintDescription: must be a valid CIDR block format.
-        Description: >-
-            The CIDR block for the subnet located in Availability Zone 2 where
-            the FortiGate Autoscale instances will be deployed to.
-    FortiGateInstanceType:
+    CustomAssetDirectory:
+        AllowedPattern: ^$|^[0-9a-zA-Z-/]*/$
+        ConstraintDescription: Deployment package S3 key prefix can include numbers, lowercase letters, uppercase letters, hyphens (-), and forward slash (/). If provided, it must end with a forward slash (/).
```
Comment (ilee-fortinet) [inline templates/autoscale-tgw-new-vpc.template.yaml:30]:
```suggestion
        ConstraintDescription: Deployment package S3 key prefix can include numbers, lowercase letters, uppercase letters, hyphens (-), and forward slashes (/). If provided, it must end with a forward slash (/).
```

File: templates/autoscale-tgw-new-vpc.template.yaml [hunk 15]
```diff
@@ -7,362 +7,119 @@ Description: >-
...
+        Type: Number
+    FgtAsgMaxSizePayg:
         ConstraintDescription: must be a valid number not less than 0.
-        Description: >-
-            Minimum number of FortiGate instances in the BYOL Auto Scaling
-            group. For specific use cases, set to 0 for On-Demand-only, and >= 2
-            for BYOL-only or hybrid licensing.
-    FgtAsgMaxSizeByol:
+        Default: 6
+        Description: Maximum number of FortiGate instances in the On-Demand Auto Scaling group. For specific use cases, set to 0 for BYOL-only, >= 2 for On-Demand-only, and >= 0 for hybrid licensing. This number must be greater than or equal to the Minimum group size (On-Demand).
```
Comment (ilee-fortinet) [inline templates/autoscale-tgw-new-vpc.template.yaml:75]:
```suggestion
        Description: Maximum number of FortiGate instances in the On-Demand Auto Scaling group. For specific use cases, set to 0 for BYOL-only, >= 2 for On-Demand-only, and >= 0 for hybrid licensing. This number must be greater than or equal to the minimum group size (On-Demand).
```

File: templates/autoscale-tgw-new-vpc.template.yaml [hunk 16]
```diff
@@ -7,362 +7,119 @@ Description: >-
...
-    UseCustomAssetLocation:
-        Type: String
-        Default: 'no'
-        AllowedValues:
-            - 'yes'
-            - 'no'
-        ConstraintDescription: must choose from the provided options.
-        Description: Will use a custom S3 location for custom assets.
-    CustomAssetContainer:
+        Description: The password for the 'Autoscale admin username'. The password must conform to the FortiAnalyzer password policy and have a min length of 8 and a max length 128. If you need to enable KMS encryption, refer to the documentation.
```
Comment (ilee-fortinet) [inline templates/autoscale-tgw-new-vpc.template.yaml:106]:
```suggestion
        Description: The password for the 'Autoscale admin username'. The password must conform to the FortiAnalyzer password policy and have a minimum length of 8 and a maximum length of 128. If you need to enable KMS encryption, refer to the documentation.
```

File: templates/autoscale-tgw-new-vpc.template.yaml [hunk 17]
```diff
@@ -7,362 +7,119 @@ Description: >-
...
+        AllowedPattern: ^$|^(([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5]){1}$
+        ConstraintDescription: must be a valid IPv4 format.
         Default: ''
-        Description: >-
-            The sub path within the 'custom asset container' which serves as the
-            top level directory of all your custom assets. If 'use custom asset
-            location' is set to 'yes', and this value is left empty, the 'custom
-            asset container' will serve as the top level directory.
-    FortiAnalyzerIntegrationOptions:
+        Description: The custom private IP address to be used by the FortiAnalyzer. Must be within the Public subnet 1 CIDR range. Required if 'FortiAnalyzer Integration' is set to 'yes'. If 'FortiAnalyzer Integration' is set to 'no', any input will be ignored.
```
Comment (ilee-fortinet) [inline templates/autoscale-tgw-new-vpc.template.yaml:120]:
```suggestion
        Description: The custom private IP address to be used by the FortiAnalyzer. Must be within the public subnet 1 CIDR range. Required if 'FortiAnalyzer Integration' is set to 'yes'. If 'FortiAnalyzer Integration' is set to 'no', any input will be ignored.
```

File: templates/autoscale-tgw-new-vpc.template.yaml [hunk 18]
```diff
@@ -376,85 +133,175 @@ Parameters:
...
+        Description: Instance type to launch as FortiAnalyzer. Instance types are available with different vCPU sizes and bandwidths. For more information, see https://aws.amazon.com/ec2/instance-types/.
         Type: String
-        Default: 6.4.7
+    FortiAnalyzerIntegrationOptions:
+        AllowedValues:
+            - 'yes'
+            - 'no'
+        ConstraintDescription: must choose from the provided options.
+        Default: 'yes'
+        Description: Choose 'yes' to incorporate FortiAnalyzer into Fortinet FortiGate Auto Scaling to use extended features that include storing logs into FortiAnalyzer.
```
Comment (ilee-fortinet) [inline templates/autoscale-tgw-new-vpc.template.yaml:145]:
```suggestion
        Description: Choose 'yes' to incorporate FortiAnalyzer into Fortinet FortiGate Auto Scaling to use extended features that include storing logs in FortiAnalyzer.
```

File: templates/autoscale-tgw-new-vpc.template.yaml [hunk 19]
```diff
@@ -376,85 +133,175 @@ Parameters:
...
+        ConstraintDescription: must be a valid number between 30 and 90.
+        Default: 30
+        Description: The length of time (in seconds) that a FortiGate instance waits between sending heartbeat requests to the Autoscale handler. Minimum is 30. Maximum is 90.
+        MaxValue: 90
+        MinValue: 30
+        Type: Number
+    HeartBeatLossCount:
+        ConstraintDescription: must be a valid number between 1 and 65535.
+        Default: 10
+        Description: Number of consecutively lost heartbeats. When the Heartbeat Loss Count has been reached, the VM is deemed unhealthy and fail-over activities will commence.
```
Comment (ilee-fortinet) [inline templates/autoscale-tgw-new-vpc.template.yaml:214]:
```suggestion
        Description: Maximum number of consecutively lost heartbeats. When the Heartbeat loss count is reached, the VM is deemed unhealthy and failover activities commence.
```

File: templates/autoscale-tgw-new-vpc.template.yaml [hunk 20]
```diff
@@ -376,85 +133,175 @@ Parameters:
...
+    PublicSubnet2Cidr:
+        AllowedPattern: ^(([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5]){1}(\/([0-9]|[1-2][0-9]|3[0-2]))?$
+        ConstraintDescription: must be a valid CIDR block format.
+        Default: 192.168.1.0/24
+        Description: The CIDR block for the subnet located in Availability Zone 2 where the FortiGate Autoscale instances will be deployed to.
+        Type: String
+    ResourceTagPrefix:
+        AllowedPattern: '[0-9a-zA-Z@.#-]+'
+        ConstraintDescription: Resource tag prefix can include numbers, lowercase letters, uppercase letters, ampersat(@) , hyphens (-), period (.), and hash (#). Max length is 50.
+        Description: The ResourceGroup Tag Key used on all resources and as the name prefix of all applicable resources. Can only contain numbers, lowercase letters, uppercase letters, ampersat(@), hyphens (-), period (.), and hash (#). Max length is 50.
```
Comment (ilee-fortinet) [inline templates/autoscale-tgw-new-vpc.template.yaml:251]:
```suggestion
        Description: The ResourceGroup tag key used on all resources and as the name prefix of all applicable resources. Can only contain numbers, lowercase letters, uppercase letters, ampersats (@), hyphens (-), periods (.), and hashes (#). Max length is 50.
```

File: templates/autoscale-tgw-new-vpc.template.yaml [hunk 21]
```diff
@@ -376,85 +133,175 @@ Parameters:
...
+    S3BucketName:
+        AllowedPattern: ^[0-9a-zA-Z]+([0-9a-zA-Z-.]*[0-9a-zA-Z])*$
+        ConstraintDescription: Deployment resource S3 bucket name can include numbers, lowercase letters, uppercase letters, period (.), and hyphens (-). It cannot start or end with a hyphen (-).
+        Description: Name of the S3 bucket that contains the FortiGate Autoscale deployment package. Can only contain numbers, lowercase letters, uppercase letters, period (.), and hyphens (-). It cannot start or end with a hyphen (-).
+        MaxLength: 63
+        MinLength: 3
+        Type: String
+    S3KeyPrefix:
+        AllowedPattern: ^$|^[0-9a-zA-Z-./]*/$
+        ConstraintDescription: Deployment package S3 key prefix can include numbers, lowercase letters, uppercase letters, hyphens (-), period (.), and forward slash (/). If provided, it must end with a forward slash (/).
```
Comment (ilee-fortinet) [inline templates/autoscale-tgw-new-vpc.template.yaml:263]:
```suggestion
        ConstraintDescription: Deployment package S3 key prefix can include numbers, lowercase letters, uppercase letters, hyphens (-), periods (.), and forward slashes (/). If provided, it must end with a forward slash (/).
```

File: docs/deployment_templates.md [hunk 22]
```diff
@@ -0,0 +1,230 @@
+# CloudFormation deployment templates
+
+There are three deployment templates provided as entry template. They are located the /templates directory. They are used in three different deployment scenarios.
+
+## Deploy Autoscale into a new VPC
+
+### Scenario
+
+The deployment template creates a new VPC and will deploy the Autoscale and related components into this new VPC.
```
Comment (ilee-fortinet) [inline docs/deployment_templates.md:9]:
```suggestion
The deployment template creates a new VPC and deploys the Autoscale and related components into this new VPC.
```

File: docs/deployment_templates.md [hunk 23]
```diff
@@ -0,0 +1,230 @@
...
+
+_Default_: The default value for this parameter.
+
+_Description_: Description of the parameter.
+
+Required parameters will have a bold font on their label column in the table below.
+
+| Name | Label | Default | Description |
+| --- | --- | --- | --- |
+| AutoscaleNotificationSubscriberEmail | Autoscale notifications subscriber email | | The email address (AWS SNS Topic subscriber) to receive Autoscale notifications. If provided, the template can only accept one email address. An email will be sent to the address to confirm the subscription. |
```
[the remark this answers is not in the document]
Comment (JaydenLiang) [reply docs/deployment_templates.md:31]: Hi @ilee-fortinet , ***AWS SNS Topic*** is a service and treated as a name. Can you please update your suggestion?

File: docs/deployment_templates.md [hunk 24]
```diff
@@ -0,0 +1,230 @@
...
```
[the remark this answers is not in the document]
Comment (JaydenLiang) [reply docs/deployment_templates.md:107]: Hi @ilee-fortinet , AWS SNS Topic is a service and treated as a name. Can you please update your suggestion?

File: docs/deployment_templates.md [hunk 25]
```diff
@@ -0,0 +1,230 @@
...
+| S3KeyPrefix | **S3 resource folder** | |  Name of the S3 folder that stores the FortiGate Autoscale deployment resources. Can only contain numbers, lowercase letters, uppercase letters, hyphens (-), period (.), and forward slashes (/). If provided, it must end with a forward slash (/). |
+| SyncRecoveryCount | Autoscale sync recovery count | 3 | Number of consecutive on-time heartbeats required for a VM to become healthy again. This parameter is only used when **Terminate unhealthy VM** is set to **no** and allows for the VM to recover from an unhealthy state. |
+| TerminateUnhealthyVm | Terminate unhealthy VM | no | Terminate any VM that is deemed unhealthy by the Autoscale. |
+| UseCustomAssetLocation | Use custom asset location | no | Set to yes to use a custom S3 location for custom assets such as licenses and customized configsets. |
+| VpcCidr | **VPC CIDR** | |  Classless Inter-Domain Routing (CIDR) block for the FortiGate Autoscale VPC. |
+| VpcEndpointId | **Private VPC Endpoint ID** | |  ID of the Private VPC Endpoint associated with the existing VPC. The Private VPC Endpoint must has enabled the **Private DNS names**. |
```
[the remark this answers is not in the document]
Comment (JaydenLiang) [reply docs/deployment_templates.md:157]: Hi @ilee-fortinet , ***Private VPC Endpoint*** is treated as a name. Can you please update your suggestion?

File: docs/deployment_templates.md [hunk 26]
```diff
@@ -0,0 +1,230 @@
...
+
+_Default_: The default value for this parameter.
+
+_Description_: Description of the parameter.
+
+Required parameters will have a bold font on their label column in the table below.
+
+| Name | Label | Default | Description |
+| --- | --- | --- | --- |
+| AutoscaleNotificationSubscriberEmail | Autoscale notifications subscriber email | | The email address (AWS SNS Topic subscriber) to receive Autoscale notifications. If provided, the template can only accept one email address. An email will be sent to the address to confirm the subscription. |
```
[the remark this answers is not in the document]
Comment (JaydenLiang) [reply docs/deployment_templates.md:186]: Hi @ilee-fortinet , ***AWS SNS Topic*** is treated as a name. Can you please update your suggestion?

File: templates/autoscale-existing-vpc.template.yaml [hunk 27]
```diff
@@ -6,504 +6,330 @@ Description: >-
...
-        AllowedPattern: '^$|^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$'
-        ConstraintDescription: must be a valid email address.
-        Description: >-
-            The email address (AWS SNS Topic subscriber) to receive Autoscale
-            notifications. If provided, the template can only accept one email
-            address. An email will be sent to the address to confirm the
-            subscription.
-    TerminateUnhealthyVm:
+    VpcEndpointId:
+        Description: ID of the Private VPC Endpoint associated with the existing VPC. The Private VPC Endpoint must has enabled the 'Private DNS names'.
```
[the remark this answers is not in the document]
Comment (JaydenLiang) [reply templates/autoscale-existing-vpc.template.yaml:328]: Hi @ilee-fortinet , ***Private VPC Endpoint*** is treated as a name. Can you please update your suggestion?

File: templates/autoscale-new-vpc.template.yaml [hunk 28]
```diff
@@ -6,510 +6,337 @@ Description: >-
...
-        Description: >-
-            Name of the S3 bucket that contains the FortiGate Autoscale
-            deployment package. Can only contain numbers, lowercase letters,
-            uppercase letters, period (.), and hyphens (-). It cannot start or
-            end with a hyphen (-).
-    S3KeyPrefix:
+    AutoscaleNotificationSubscriberEmail:
+        AllowedPattern: ^$|^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$
+        ConstraintDescription: must be a valid email address.
+        Description: The email address (AWS SNS Topic subscriber) to receive Autoscale notifications. If provided, the template can only accept one email address. An email will be sent to the address to confirm the subscription.
```
[the remark this answers is not in the document]
Comment (JaydenLiang) [reply templates/autoscale-new-vpc.template.yaml:12]: Hi @ilee-fortinet , ***AWS SNS Topic*** is treated as a name. Can you please update your suggestion?

File: templates/autoscale-tgw-new-vpc.template.yaml (hunk 12 above)
Comment (JaydenLiang) [reply templates/autoscale-tgw-new-vpc.template.yaml:13]: Hi @ilee-fortinet , ***AWS SNS Topic*** is treated as a name. Can you please update your suggestion?

File: templates/autoscale-tgw-new-vpc.template.yaml (hunk 13 above)
Comment (JaydenLiang) [reply templates/autoscale-tgw-new-vpc.template.yaml:20]: This is treated as a name. no need to change.

File: docs/deployment_templates.md (hunk 23 above)
[the remark this answers is not in the document]
Comment (ilee-fortinet) [reply docs/deployment_templates.md:31]: Updated

File: docs/deployment_templates.md (hunk 24 above)
[the remark this answers is not in the document]
Comment (ilee-fortinet) [reply docs/deployment_templates.md:107]: Updated

File: docs/deployment_templates.md (hunk 25 above)
[the remark this answers is not in the document]
Comment (ilee-fortinet) [reply docs/deployment_templates.md:157]: Updated

File: docs/deployment_templates.md (hunk 26 above)
[the remark this answers is not in the document]
Comment (ilee-fortinet) [reply docs/deployment_templates.md:186]: Updated

File: templates/autoscale-existing-vpc.template.yaml (hunk 27 above)
[the remark this answers is not in the document]
Comment (ilee-fortinet) [reply templates/autoscale-existing-vpc.template.yaml:328]: Updated

File: templates/autoscale-tgw-new-vpc.template.yaml (hunk 12 above)
Comment (ilee-fortinet) [reply templates/autoscale-tgw-new-vpc.template.yaml:13]: Updated

**Review by ilee-fortinet — COMMENTED** [review]
Fixed capitalization in comments per Jayden's request

**Pull request merged**